### PR TITLE
 adding virtual copy and chap folder modifications

### PR DESCRIPTION
**Pull request by leenaj0hn**
1. Virtual copy testcases
2. Folder modifications for chap

### CODE REVIEW

File: hpe_3par_kubernetes_manager.py [hunk 1]
```diff
@@ -819,9 +843,9 @@ def verify_volume_properties_primera(hpe3par_volume, **kwargs):
             if 'size' in kwargs:
                 if hpe3par_volume['sizeMiB'] != int(kwargs['size']) * 1024:
                     return False
-            else:
-                if hpe3par_volume['sizeMiB'] != 102400:
-                    return False
+                else:
```
Comment (saranke) [inline hpe_3par_kubernetes_manager.py:846]: Changes in indentation makes else of inner if condition, whereas before changes it was for outer if. Request to confirm if this change is intensional.

**Review by saranke — COMMENTED** [review]
Added a comment to a change in manager.py.
Please reply to the comment.

File: hpe_3par_kubernetes_manager.py (hunk 1 above)
Comment (leenaj0hn) [reply hpe_3par_kubernetes_manager.py:846]: Yes, this was intentional as the subsequent volume property checks were driven based on the "provisioning" option being available in kwargs. This did not work for snapshot volume as the provisioning type was not relevant. So to make it common across base and snapshot volume, made the change to check each individual property if present in kwargs

Comment (saranke) [reply hpe_3par_kubernetes_manager.py:846]: Thanks for confirmation. Approving the PR.

**Pull request merged**